gateway/src/apicast/policy/maintenance_mode/maintenance_mode.lua

```diff
@@ -19,20 +26,100 @@ function _M.new(configuration)
   policy.message_content_type = default_message_content_type
 
   if configuration then
+    policy.maintenance_upstreams = configuration.upstreams
     policy.status_code = tonumber(configuration.status) or policy.status_code
     policy.message = configuration.message or policy.message
     policy.message_content_type = configuration.message_content_type or policy.message_content_type
   end
 
+  policy:load_condition(configuration)
   return policy
 end
 
-function _M:rewrite()
-  ngx.header['Content-Type'] = self.message_content_type
-  ngx.status = self.status_code
-  ngx.say(self.message)
+function _M:load_condition(config)
+  if not config or not config.condition then
+    return
+  end
 
+  local operations = {}
+  for _, operation in ipairs(config.condition.operations or {}) do
+    table.insert( operations,
+      Operation.new(
+        operation.match, operation.match_type,
+        operation.op,
+        operation.value, operation.value_type or default_template_type))
+  end
+  self.condition = Condition.new( operations, config.condition.combine_op or default_combine_op)
+end
+
+local function maintenance_mode(policy)
+  ngx.log(ngx.DEBUG, 'Maintenance mode enabled for this request')
+  ngx.header['Content-Type'] = policy.message_content_type
+  ngx.status = policy.status_code
+  ngx.say(policy.message)
   return ngx.exit(ngx.status)
 end
 
+local function uris_match(uri1, uri2)
+  local scheme1 = uri1.scheme or "http"
+  local scheme2 = uri2.scheme or "http"
+  local port1 = uri1.port or port1
+  if not port1 then
+    port1 = (scheme1 == "https" and 443 or 80)
+  end
+  local port2 = uri2.port or port2
+  if not port2 then
+    port2 = (scheme2 == "https" and 443 or 80)
+  end
+
+  local path1 = uri1.path or ""
+  path1 = string.gsub(path1, "/$", "")
+  local path2 = uri2.path or ""
+  path2 = string.gsub(path2, "/$", "")
+
+  return scheme1 == scheme2 and uri1.host == uri2.host and port1 == port2 and path1 == path2 
```

Comment:
Why are you checking the path too?

Reply:
@palmieric fixed it, I was supposed to "match" the paths instead of using ==. Thanks!